src/components/DiagramFrame/MetadataPanel/MetadataPanel.tsx

```diff
@@ -0,0 +1,196 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2020 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import React from 'react';
+import {
+  Badge,
+  ButtonTransparent,
+  Code,
+  Flex,
+  FlexItem,
+  Heading,
+  SpaceVertical,
+  Paragraph,
+  Icon,
+  Space,
+  Tabs,
+  Tab,
+  TabList,
+  TabPanel,
+  TabPanels,
+  theme
+} from "@looker/components"
+import { ILookmlModel, ILookmlModelExplore } from "@looker/sdk"
+import { ExternalLink } from "../../ExternalLink"
+import JoinIcon from "./JoinIcon"
+import MetadataPanelTable from "./MetadataPanelTable"
+import {MetadataInfoPanel} from "./MetadataInfoPanel"
+import {MetadataFooter} from "./MetadataFooter"
+import {LookmlCodeBlock} from "./LookmlCodeBlock"
+import { METADATA_PANEL_PIXEL } from "../../../utils/constants"
+import { LookmlObjectMetadata, SelectionInfoPacket } from "../../interfaces"
+import { ILookmlModelExploreField, ILookmlModelExploreJoins } from '@looker/sdk/lib/sdk/3.1/models';
+import { getFields } from '../../../utils/diagrammer'
+import { exploreFieldURL } from '../../../utils/urls'
+import { useExplore } from '../../../utils/fetchers'
```

Comment:
A nice pattern we use is to have an `index.ts` file in each folder that hoists the exports up to a single location for a clearer interface. Then you can have
`import { getFields, exploreFieldURL, useExplore } from '../../../utils` instead of three separate lines.
Same goest for your components etc. all throughout this PR

Also a better alternative to what you have below which is a single "utils" file with several utils in it. You might instead want a utils folder with a file called "getMetadata.ts" that exports those up to the utils index.ts file

Reply:
https://looker.atlassian.net/browse/MRKTI-474